modules/client/test/integration/admin/methods.test.ts

```diff
@@ -0,0 +1,260 @@
+    const daoCreation = await deployContracts.createAdminDAO(
+      deployment,
+      "testDAO",
+      TEST_WALLET_ADDRESS,
+    );
+    pluginAddress = daoCreation.pluginAddrs[0];
+  });
+
+  afterAll(async () => {
+    await server.close();
+  });
+
+  describe("Proposal Creation", () => {
+    it("Should create a new proposal locally", async () => {
+      const ctx = new Context(contextParamsLocalChain);
+      const ctxPlugin = ContextPlugin.fromContext(ctx);
+      const adminClient = new ClientAdmin(ctxPlugin);
+      const client = new Client(ctx);
+
+      // generate actions
+      const action = await client.encoding.withdrawAction(pluginAddress, {
+        recipientAddress: "0x1234567890123456789012345678901234567890",
+        amount: BigInt(1),
+        reference: "test",
+      });
+
+      const metadata: ProposalMetadata = {
+        title: "Best Proposal",
+        summary: "this is the sumnary",
+        description: "This is a very long description",
+        resources: [
+          {
+            name: "Website",
+            url: "https://the.website",
+          },
+        ],
+        media: {
+          header: "https://no.media/media.jpeg",
+          logo: "https://no.media/media.jpeg",
+        },
+      };
+
+      const ipfsUri = await adminClient.methods.pinMetadata(metadata);
+
+      const proposalParams: ExecuteProposalParams = {
+        pluginAddress,
+        metadataUri: ipfsUri,
+        actions: [action],
+      };
+
+      for await (
+        const step of adminClient.methods.executeProposal(
+          proposalParams,
+        )
+      ) {
+        switch (step.key) {
+          case ExecuteProposalStep.EXECUTING:
+            expect(typeof step.txHash).toBe("string");
+            expect(step.txHash).toMatch(/^0x[A-Fa-f0-9]{64}$/i);
+            break;
+          case ExecuteProposalStep.DONE:
+            break;
+          default:
+            throw new Error(
+              "Unexpected proposal execution step: " +
+                Object.keys(step).join(", "),
+            );
+        }
+      }
+    });
+  });
+
+  describe("Data retrieval", () => {
+    it("Should fetch the given proposal", async () => {
+      const ctx = new Context(contextParams);
+      const ctxPlugin = ContextPlugin.fromContext(ctx);
+      const client = new ClientAdmin(ctxPlugin);
+
+      const proposalId = TEST_ADMIN_PROPOSAL_ID;
+
+      mockedIPFSClient.cat.mockResolvedValue(
+        Buffer.from(
+          JSON.stringify({
+            title: "Title",
+            summary: "Summary",
+            description: "Description",
+            resources: [{
+              name: "Name",
+              url: "URL",
+            }],
+          }),
+        ),
+      );
+
+      const proposal = await client.methods.getProposal(proposalId);
+
+      expect(typeof proposal).toBe("object");
+      expect(proposal === null).toBe(false);
+      if (proposal) {
```

Comment:
Isn't it simpler to do `if(!proposal) throw new Error()`? No need for the reverse if